plugins/inputs/smart/smart.go

```diff
@@ -373,6 +391,37 @@ func gatherDisk(acc telegraf.Accumulator, timeout internal.Duration, usesudo, co
 				}
 			}
 		}
+
+		if m.ErrorCounterLog {
+			ecl := errorCounterLog.FindStringSubmatch(line)
+			if len(ecl) == 9 {
+				eclTags := map[string]string{}
+				eclFields := make(map[string]interface{})
+
+				eclTags["page_name"] = ecl[1]
+
+				keys := [...]string{"device", "model", "serial_no", "wwn", "capacity", "enabled"}
+				for _, key := range keys {
+					if value, ok := deviceTags[key]; ok {
+						eclTags[key] = value
+					}
+				}
+
+				eclFields["errors_corrected_by_eccfast"] = ecl[2]
+				eclFields["errors_corrected_by_eccdelayed"] = ecl[3]
+				eclFields["errors_corrected_by_rereads_rewrites"] = ecl[4]
+				eclFields["total_errors_corrected"] = ecl[5]
+				eclFields["correction_algorithm_invocations"] = ecl[6]
```

Comment:
As the metric already contains an `error` label, should we maybe remove the `errors` part from the field name?

Reply:
@srebhan idea was to keep the names same as returned by smartctl so it's easier to correlate to the output of the tool. However I can see the issues raised by you in the other comment next comment regarding "gigabytes_processed".

Reply:
Understood. Not sure though if it is really a good idea to track the smartctl names. What if they change the output name? Will you change the field? ;-)